File: king-admin-java/src/main/java/com/oukingtim/web/SysRoleController.java
package com.oukingtim.web;


import com.oukingtim.domain.SysRole;
import com.oukingtim.service.SysRoleService;
import com.oukingtim.web.vm.ResultVM;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * Created by oukingtim
 */
@RestController
@RequestMapping("/sys/role")
public class SysRoleController extends BaseController<SysRoleService, SysRole> {

    /**
     * 获取角色集合
     */
    @GetMapping("/getlist")
    public ResultVM getList(@RequestParam(required = false) String userId) {
        List<SysRole> roleList = service.getList(userId);
        return ResultVM.ok(roleList);
    }
}

File: king-admin-kotlin/src/main/kotlin/com/oukingtim/web/SysRoleController.kt
package com.oukingtim.web

import com.oukingtim.domain.SysRole
import com.oukingtim.service.SysRoleService
import com.oukingtim.web.vm.ResultVM
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

/**
 * Created by oukingtim
 */
@RestController
@RequestMapping("/sys/role")
class SysRoleController : BaseController<SysRoleService, SysRole>() {

    /**
     * 获取角色集合
     */
    @GetMapping("/getlist")
    fun getList(@RequestParam(required = false) userId: String) = ResultVM.ok(service!!.getList(userId))

}